packages/testing/src/presentation-testing/InternalUtils.ts

```diff
@@ -27,19 +27,36 @@ export function setTestOutputDir(directoryPath: string | undefined) {
  * - Resolving the output file name under the known test output directory
  * - Making directories as necessary
  * - Removing a previous copy of the output file
- * @param fileName Name of output file
+ * @param fileName Name of output file. The actual file name may get modified to fit within the file system limits.
  * @internal
  */
 export function setupOutputFileLocation(fileName: string): LocalFileName {
   const outputDirectoryPath = getTestOutputDir();
   !IModelJsFs.existsSync(outputDirectoryPath) && IModelJsFs.mkdirSync(outputDirectoryPath);
 
-  const outputFile = path.join(outputDirectoryPath, `${fileName}.bim`);
-  IModelJsFs.existsSync(outputFile) && IModelJsFs.unlinkSync(outputFile);
-  return outputFile;
+  const outputFilePath = limitFilePathLength(path.join(outputDirectoryPath, `${fileName}.bim`));
+  IModelJsFs.existsSync(outputFilePath) && IModelJsFs.unlinkSync(outputFilePath);
+  return outputFilePath;
 }
 
 /** @internal */
 export function createFileNameFromString(str: string) {
   return sanitize(str.replace(/[ ]+/g, "-").replaceAll("`", "").replaceAll("'", "")).toLocaleLowerCase();
 }
+
+/** @internal */
+export function limitFilePathLength(filePath: string) {
+  const { dir, name, ext } = path.parse(filePath);
+
+  const allowedFileNameLength = 260 - 12 - 1 - (dir.length + 1) - ext.length;
+  if (allowedFileNameLength <= 0) {
```

Comment:
It has to be at least 3 symbols since shortened name has `...` in it

Reply:
should be okay now